src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+        """
+        if not self.check_model_version_exists(name, version):
+            raise KeyError(
+                f"The model version with name '{name}' and version '{version}' "
+                "does not exist."
+            )
+        # Update the model description.
+        if version_description:
+            try:
+                self.mlflow_client.update_model_version(
+                    name=name,
+                    version=version,
+                    description=version_description,
+                )
+            except MlflowException as e:
+                logger.error(
+                    f"Failed to update the description of model version "
+                    f"'{name}:{version}' in the MLflow model registry."
+                )
+                raise e
+        # Update the model tags.
+        if version_tags:
+            try:
+                for key, value in version_tags.items():
+                    self.mlflow_client.set_model_version_tag(
+                        name=name,
+                        version=version,
+                        key=key,
+                        value=value,
+                    )
+            except MlflowException as e:
+                logger.error(
+                    f"Failed to update the tags of model version "
+                    f"'{name}:{version}' in the MLflow model registry."
+                )
+                raise e
+        # Update the model stage.
+        if version_stage:
+            try:
+                self.mlflow_client.transition_model_version_stage(
+                    name=name,
+                    version=version,
+                    stage=version_stage.value,
+                )
+            except MlflowException as e:
+                logger.error(
+                    f"Failed to update the stage of model version "
+                    f"'{name}:{version}' in the MLflow model registry."
+                )
+                raise e
+        return self.get_model_version(name, version)
+
+    def get_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> ModelVersion:
+        """Get a model version from the MLflow model registry.
+
+        Args:
+            name (str): The name of the model.
+            version (str): The version of the model.
+
+        Raises:
+            MlflowException: If mlflow fails to get the model version.
+
+        Returns:
+            The model version.
+        """
+        # Get the model version from the MLflow model registry.
+        try:
+            mlflow_model_version = self.mlflow_client.get_model_version(
+                name=name,
+                version=version,
+            )
+        except MlflowException as e:
+            logger.error(
+                f"Failed to get model version '{name}:{version}' from the "
+                "MLflow model registry."
+            )
+            raise e
+        # Return the model version.
+        return ModelVersion(
```

Comment:
This code of converting from an mlflow model version to ours happens in a lot of the methods here, we can probably extract that into a separate one to avoid lots of the duplicate code?